### ISSUE TITLE
Search by date, multiple tag, location,.....

### ISSUE THREAD

**Issue by iwanttobefreak** [nextcloud/photos#445]
Hello! First of all, thanks for app.
I want to do a complex search of photos.
For example: I go to a place every year with different people and to do diferent things (with different tags)
Search like:
date: beetwen 2008-2009
place: Madrid
people: Laia
event: holidays (or work....)

Thanks!

**Comment by Mikescops** [comment]
Just wondering is this would better fit the global files search of nextcloud/server :thinking:

**Comment by jancborchardt** [comment]
This is very related to the albums discussion, and also goes into the topic of "automatic album creation". Cause you don’t really want to create this filter every single time, or think about what you need to search for.

What’s nice in iOS Photos is that they directly create virtual albums for any trip. They group photos by time and location and thus most of the time you don’t need to fiddle with manual album creation. They call it "Memories": https://support.apple.com/en-us/HT207023

Of course this might be a bigger thing to look into, but it will also fix a lot of UX problems where currently people have to do manual work of searching and categorization.

**Comment by Spartachetto** [comment]
My impression is that the proposal from @jancborchardt is a really good starting point.
I mean that it will cover a lot of necessities for complex queries but not all, so having it would be really useful but, on a long time scale, I think more would be better.

Let me explain my reasoning:
1. Time and place is not enough. Usually I group vacation photos by myself. The more difficult to manage are the everyday photos: I distinguish between family photos and the interesting landscape that I see when going around. To distinguish those you need more than time and place, also for some photos in the end the criteria is just my personal taste....
2. Sometimes I happened to look for a photo that I remembered. I remembered the place, the period of the year (autumn), maybe other things but **not** the year.

To solve case 2 I'd love an advanced search system (place = Berlin, months = from September to November, ...). Probably the most interesting thing would be before to implement such search system in the server (so I can also search for that document of which I remember just something....) and then to define a graphical interface for photos.

I know, I am asking for a **really big** thing :sweat_smile: . Maybe you could consider if you envision this as the final destination....

Just a last clarification. I guess that this is complementary to the AI classification of photos. In case that functionality becomes available I could ask for the photos that fit in place = Berlin, months = September to November, subject = tree, ....  where the subject categorization comes from AI.

**Comment by jancborchardt** [comment]
Let’s always remember we are designing Nextcloud for non-technical people. People who don’t want to or don’t necessarily have the time to come up with an organization for their photos, who just want it to work and for that reason buy Apple products.

For people who create folders themselves, the current implementation already works.

**Comment by Spartachetto** [comment]
_For people who create folders themselves, the current implementation already works._ 

I understand your point and I sort of agree. I was just suggesting possible improvements for users like me. 

When you wrote _ might be a bigger thing to look into_, I understood that the suggested improvements could be a bit broader than just "do like Apple". Just trying to help with feedback.

**Comment by Spartachetto** [comment]
I realized that what I was proposing already exists. Nextcloud has such functionality:

**Comment by dmuensterer** [comment]
Is there any progress on this?
I just started using Nextcloud as an alternative to iCloud which confronted me with the following use case:

I have all photos synced with Nextcloud. Now I want to create a folder to share with friends with all photos taken on May 5th 2021.
How to go about that?

**Comment by xermus** [comment]
I totally agree that Nextcloud asap needs a better SEARCH functionality where you can lookup files by TYPE, DATE, UPLOADER, etc. Just having a dedicated SEARCH icon in the top navication would be a fantastic solution, because SEARCHING what you uploaded is of course one of the main things which should be accompished.

Right now the SEARCH is not good, as I have hard times finding ie. ".mp4" files created before 2019 etc.

**Comment by mgscreativa** [comment]
Hi! really need to do search of files by date range in a given folder, is that even possible?

**Comment by eibex** [comment]
I am still using maps for this but it is painfully slow to load all pictures on the map.
Given how Nextcloud's recognise app works, would it be feasible to sort out the "location" part of this issue by adding a tag of the location (e.g. country) by parsing exif data? Not sure if this is in scope for the photos app though.

**Comment by JPustkuchen** [comment]
Run into this and especially not being able to filter by date range is kind of horrible, if you know when the picture was taken, but you have to scroll down to the year and month manually... takes minutes of wasted time ;D

Thank you for this suggestion!

**Comment by nemhods** [comment]
I have another use-case for this. I want to sort my photos into albums / events / occasions. I import a bunch of miscellaneous photos and then have to sort this into albums. I want no photo to be unsorted, so I would need a filter like "Photos NOT assigned to an album".
